src/sagas/import-project.saga.test.js

```diff
@@ -0,0 +1,280 @@
+import { call, put, cancel, select, takeEvery } from 'redux-saga/effects';
+import rootSaga, {
+  handlePathInput,
+  showImportDialog,
+  handleImportError,
+  importProject,
+  inferProjectType,
+} from './import-project.saga';
+import electron from 'electron';
+import {
+  importExistingProjectStart,
+  importExistingProjectError,
+  importExistingProjectFinish,
+  SHOW_IMPORT_EXISTING_PROJECT_PROMPT,
+  IMPORT_EXISTING_PROJECT_START,
+} from '../actions';
+import {
+  loadPackageJson,
+  writePackageJson,
+} from '../services/read-from-disk.service';
+import { getInternalProjectById } from '../reducers/projects.reducer';
+import { getColorForProject } from '../services/create-project.service';
+
+jest.mock('electron', () => ({
+  remote: { dialog: { showOpenDialog: jest.fn(), showErrorBox: jest.fn() } },
+}));
+
+describe('import-project saga', () => {
+  const { dialog } = window.require('electron').remote;
+
+  describe('handlePathInput saga', () => {
+    it('should do nothing if argument paths is undefined', () => {
+      const saga = handlePathInput();
+      expect(saga.next().value).toEqual(cancel());
+    });
+
+    it('should get only first path from passed array and start importing', () => {
+      const paths = ['path/to/first', 'path/to/second'];
+      const saga = handlePathInput(paths);
+      expect(saga.next().value).toEqual(
+        put(importExistingProjectStart('path/to/first'))
+      );
```

Comment:
I'm not super familiar with saga testing - wouldn't we want to test that it's equal to a plain object?

```js
const expectedValue = importExistingProjectStart('path/to/first');
// { type: IMPORT_EXISTING_PROJECT_START, path: 'path/to/first' }
 
expect(saga.next().value).toEqual(expectedValue);
```

In other words, shouldn't we check `importExistingProjectStart()` instead of `put(importExistingProjectStart())`?

Reply:
we need to check that saga will dispatch action(effect put), in other words, we expect
```javascript
expect(saga.next().value).toEqual(put({ type: IMPORT_EXISTING_PROJECT_START, path: 'path/to/first' }));
```

Reply:
ahhh, right. Awesome!